### PR TITLE
Fix PyTorch compilation on Apple M1

### LINKED ISSUE #48145: Enable PyTorch compilation on Apple Silicon
Currently PyTorch can not be compiled natively on Apple Silicon, because it is reported as "arm64" architecture and many third-party libraries only support ARMv8 or aarch64

cc @malfet @seemethere @walterddr

### PR DESCRIPTION
**Pull request by malfet**
Update cpuinfo and sleef to contain build fixes for M1

Fixes https://github.com/pytorch/pytorch/issues/48145

### CODE REVIEW

File: cmake/Modules/FindARM.cmake [hunk 1]
```diff
@@ -41,9 +41,15 @@ IF(CMAKE_SYSTEM_NAME MATCHES "Linux")
...
 ELSEIF(CMAKE_SYSTEM_NAME MATCHES "Darwin")
+   EXEC_PROGRAM("/usr/sbin/sysctl -n hw.optional.arm64" OUTPUT_VARIABLE
+      IS_ARM64)
+   IF(IS_ARM64 STREQUAL "1")
+      set(NEON_FOUND true CACHE BOOL "NEON available on ARM64")
+   ENDIF()
    EXEC_PROGRAM("/usr/sbin/sysctl -n machdep.cpu.features" OUTPUT_VARIABLE
       CPUINFO)
 
+   IF(NOT CPUINFO STREQUAL "")
```
Comment (walterddr) [inline cmake/Modules/FindARM.cmake:52]: nit: indent?

File: cmake/Modules/FindARM.cmake [hunk 2]
```diff
@@ -41,9 +41,15 @@ IF(CMAKE_SYSTEM_NAME MATCHES "Linux")
    ENDIF (OMAP4_TRUE)
 
 ELSEIF(CMAKE_SYSTEM_NAME MATCHES "Darwin")
+   EXEC_PROGRAM("/usr/sbin/sysctl -n hw.optional.arm64" OUTPUT_VARIABLE
+      IS_ARM64)
+   IF(IS_ARM64 STREQUAL "1")
+      set(NEON_FOUND true CACHE BOOL "NEON available on ARM64")
```
Comment (walterddr) [inline cmake/Modules/FindARM.cmake:47]: question: if we are setting NEON_FOUND here, we no longer need the checks below, yes?

**Review by walterddr — APPROVED** [review]

**Comment by malfet** [comment]
ASAN failures can be fixed by the following change to sleef library: https://github.com/shibatch/sleef/pull/355

File: cmake/Modules/FindARM.cmake (hunk 2 above)
Comment (malfet) [reply cmake/Modules/FindARM.cmake:47]: Correct, but we'll skip those by checking by `IF(NOT CPUINFO STREQUAL "")` check

**Comment by danieldk** [comment]
Built libtorch with this PR and tested some transformer networks through the libtorch Rust binding. Works great so far!

**Pull request closed without merging**